src/main.rs

```diff
@@ -295,6 +295,7 @@ fn notmain() -> Result<i32, anyhow::Error> {
         }
 
         eprintln!("resetting device");
+        core.set_hw_breakpoint(vector_table.hard_fault)?;
```

Comment:
Just in case a device with 0 HW breakpoints is encountered, we could query `get_available_breakpoint_units` and print a warning when that returns 0.